wheatley/bot.py

```diff
@@ -181,11 +184,13 @@ def _on_go(self) -> None:
 
     def _on_bob(self) -> None:
         """ Callback called when a user calls 'Bob'. """
-        self.row_generator.set_bob()
+        if not self._is_calling:
```

Comment:
Convincing myself this is probably the right behaviour. This is all theoretical, the only calling row gen we have is atm is complib which ignores them currently anyway!

I think we can assume any row generator that emits a call of `Bob` should have called `set_bob() on itself if needed.

If another ringer hits the bob key while ringingthen for other ringers this is indistinguishable from Wheatley, so they would either ring it / fire out / ignore it for being halfway through a  lead (this is also a weakness in Wheatley's current long memory of a bob has been called).

Some things we could do
1) Pass it to the row generator, telling it we're supposed to be calling and let it deal with it? Seems a complex for only a little hypothetical gain.
2) Emit a call of `Belay that call` (if we haven't called it ourselves) to stop humans ringing it. More fun but I suspect this might also cause a fire out...

Reply:
To be honest, I also think that this is also not the best solution.  I mainly made this change to stop Wheatley spamming the log with error messages.

I think a better solution would be to 'stash' outgoing calls and then cancel them as they come back in, so that if a row generator calls a call (e.g. `Bob`), the `Bot` class stores this and then the first `Bob` it sees coming back will cancel with the one it stored.  This would prevent row generators from hearing their own calls, but allow them to hear other ringers' calls.

What do you reckon?

Reply:
What's the error message here? I'm surprised if `set_bob()` throws (or even logs?), or causes an error by setting `self._has_bob`

I guess the current possible use case is method ringing with Wheatley saying Go/That's all and someone else putting in bobs? Seems a bit odd but reasonable to pass the bobs through.
As `set_bob()` is idempotent (until a bob occurs in the lead), we should be able to be call it multiple times and just pass all calls on without tracking those if called by the bot or not?
(I still kind of want the `Belay that call` option)

Reply:
I can't remember what I was trying to avoid, but I agree that it doesn't make sense.  I'll remove the if statements.